isis/src/base/apps/csminit/csminit.cpp

```diff
@@ -417,6 +417,37 @@ namespace Isis {
     blobLabel += PvlKeyword("PluginName", pluginName);
     cube->write(csmStateBlob);
 
+    // Try to attach rotations
+    // If a camera fails to generate the original BodyRotation
+    // table will overwrite this one (if there was an original
+    // BodyRotation table)
+    if (ui.WasEntered("ISD")) {
+      QString isdFilePath = ui.GetFileName("ISD");
+      std::ifstream isdFileStream(isdFilePath.toStdString());
+      json isd = json::parse(isdFileStream);
```

Comment:
We need to check for a parse error here. If the ISD passed in is not JSON formatted, it should fail and continue with the rest of csminit. This whole block could be wrapped in a try/catch or we could look explicitly for things like, not being able to parse it as a json and not finding the body_rotation.